app/uk/gov/nationalarchives/omega/editorial/forms/EditSetRecordFormValues.scala

```diff
@@ -53,12 +53,12 @@ object EditSetRecordFormValues {
       coveringDates = formValues.coveringDates,
       formerReferenceDepartment = formValues.formerReferenceDepartment,
       formerReferencePro = formValues.formerReferencePro,
-      startDateDay = formValues.startDateDay,
-      startDateMonth = formValues.startDateMonth,
-      startDateYear = formValues.startDateYear,
-      endDateDay = formValues.endDateDay,
-      endDateMonth = formValues.endDateMonth,
-      endDateYear = formValues.endDateYear,
+      startDateDay = String.valueOf(formValues.startDateDay.toInt),
```

Comment:
If I revert these changes, all of the current tests pass.

Manually, I see that leading zeros are preserved, so I understand the value here.

But perhaps we're missing some scenerios at the controller level (`EditSetRecordControllerSpec`)?

Reply:
Tests added in `EditSetRecordControllerSpec`

Reply:
Good. I like the arrangements.

But I do have a few concerns noted in other comments.